bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/AndroidDebugBridgeDevice.java

```diff
@@ -0,0 +1,285 @@
+        try {
+            File directory = new File(ADB_FOLDER);
+            if (!directory.exists()) {
+                directory.mkdir();
+            }
+            adbCrypto = loadKeyPair(ADB_FOLDER + File.separator + "adb_pub.key",
+                    ADB_FOLDER + File.separator + "adb.key");
+        } catch (NoSuchAlgorithmException | IOException | InvalidKeySpecException e) {
+            logger.warn("Unable to setup adb keys: {}", e.getMessage());
+        }
+    }
+
+    private String ip = "127.0.0.1";
+    private int port = 5555;
+    private @Nullable Socket socket;
+    private @Nullable AdbConnection connection;
+
+    public void configure(String ip, int port) {
+        this.ip = ip;
+        this.port = port;
+    }
+
+    public void sendKeyEvent(String eventCode)
+            throws IOException, InterruptedException, AndroidDebugBridgeDeviceException {
+        runAdbShell("input", "keyevent", eventCode);
+    }
+
+    public void sendText(String text) throws IOException, AndroidDebugBridgeDeviceException, InterruptedException {
+        runAdbShell("input", "text", URLEncoder.encode(text, StandardCharsets.UTF_8));
+    }
+
+    public void startPackage(String packageName) throws IOException, AndroidDebugBridgeDeviceException,
+            InterruptedException, AndroidDebugBridgeDeviceException {
+        var out = runAdbShell("monkey", "--pct-syskeys", "0", "-p", packageName, "-v", "1");
+        if (out.contains("monkey aborted"))
+            throw new AndroidDebugBridgeDeviceException("Unable to open package");
+    }
+
+    public void stopPackage(String packageName)
+            throws IOException, AndroidDebugBridgeDeviceException, InterruptedException {
+        runAdbShell("am", "force-stop", packageName);
+    }
+
+    public String getCurrentPackage() throws IOException, AndroidDebugBridgeDeviceException, InterruptedException,
+            AndroidDebugBridgeDeviceReadException {
+        var out = runAdbShell("dumpsys", "window", "windows", "|", "grep", "mFocusedApp");
+        var targetLine = Arrays.stream(out.split("\n")).findFirst().orElse("");
+        var lineParts = targetLine.split(" ");
+        if (lineParts.length >= 2) {
+            var packageActivityName = lineParts[lineParts.length - 2];
+            if (packageActivityName.contains("/"))
+                return packageActivityName.split("/")[0];
+        }
+        throw new AndroidDebugBridgeDeviceReadException("can read package name");
+    }
+
+    public boolean isPlayingMedia() throws IOException, AndroidDebugBridgeDeviceException, InterruptedException {
+        // Try to get players info from audio dump
+        String audioDump = runAdbShell("dumpsys", "audio", "|", "grep", "ID:");
+        if (audioDump.length() == 0) {
+            // Fallback to media session dump
+            String devicesResp = runAdbShell("dumpsys", "media_session", "|", "grep", "PlaybackState");
```

Comment:
I personally would check on "media_sesssion" first before "audio". At least for Android TV or Fire TV. Wdyt?

In general I am not very happy with the `isPlaying()` detection. E.g. My Fire TV Stick is playing all the time, event when I am on the home screen. Maybe we can add a configuration parameter to exclude specific, user-defined packages (e.g. "com.amazon.tv.launcher") for.

IIRC I somewhere read about kind of "state_detection_rules" for Android TV:

```
state_detection_rules = {
    'com.amazon.tv.launcher': ['idle'],
    'com.netflix.ninja': ['media_session_state'],
    'com.ellation.vrv': ['audio_state'],
    'com.hulu.plus': [{'playing': {'wake_lock_size' : 4}},{'paused': {'wake_lock_size': 2}}],
    'com.plexapp.android': [{'paused': {'media_session_state˓→': 3, 'wake_lock_size': 1}},{'playing': {'media_session_state˓→': 3}},'idle']
}
```

```
VALID_STATES = ('idle', 'off', 'playing', 'paused', 'standby')
```

Reply:
I wasn't taking in account the current app for the playing check, now it should work better. 
Also I ended up implementing different ways of detect the player state, I have added the documentation. It's something similar to the solution implemented by HA. Basically you can configure a different detection mode by package name.

Reply:
Did you test them? What is your personal feeling?